x/evm/keeper/statedb.go

```diff
@@ -113,28 +113,25 @@ func (k *Keeper) ForEachStorage(ctx sdk.Context, addr common.Address, cb func(ke
 func (k *Keeper) SetBalance(ctx sdk.Context, addr common.Address, amount *big.Int) error {
 	cosmosAddr := sdk.AccAddress(addr.Bytes())
 
-	baseDenom := config.GetDenom()
+	coin, err := k.bankWrapper.GetEVMCoinBalance(ctx, cosmosAddr)
+	if err != nil {
+		return err
+	}
 
-	coin := k.bankKeeper.GetBalance(ctx, cosmosAddr, baseDenom)
+	baseDenom := config.GetEVMCoinDenom()
```

Comment:
Nit: the base denom is only used to make the coins for the `MintCoinsToAccount` and `BurnCoinsFromAccount`. We can do this internally in these functions so it is cleaner. We can pass only the amount and internally when calling the bank keeper use the evmDenom